src/mbgl/util/thread_context.cpp

```diff
@@ -1,30 +1,95 @@
 #include <mbgl/util/thread_context.hpp>
 
+#include <mbgl/util/thread_local.hpp>
+
+#include <cassert>
+
+namespace {
+
+using namespace mbgl::util;
+static tls<ThreadContext>& current = *new tls<ThreadContext>;
```

Comment:
Any particular reason to `new` this and then dereference it, as opposed to using a static value? The current implementation is technically a memory leak.